api/receiver/receive_results.go

```diff
@@ -98,38 +107,65 @@ func handleURLPayload(a AppEngineAPI, uploader string, urls []string, extraParam
 
 	var payloadType string
 	gcs := make([]string, 0, len(urls))
+	errors := make(chan error, len(urls))
```

Comment:
I would advocate for one of two strategies here:

1. Move `errors`, `wg` into the `if len(urls) > 1` block, or
2. Drop the `if len(urls) > 1` block and wait on a single goroutine.

I think I prefer 2. One complication with 1 is that `saveFileToGCS` may need to support a `nil` `sync.WaitGroup`.

Reply:
The purpose of the `if` block here is to use different schemas for `gcsPath`. When we have multiple reports for a run, the path is one level deeper for easier debugging. I rejigged the code a bit; ptal if it's better now.